### PR TITLE
update api description to be inline with the expected behaviour

### PR DESCRIPTION
**Pull request by rewantsoni**
# Describe what this PR does #

The omap generator is not deployed by default but the API states that it is. update the API description to the expected behaviour

## Related issues ##

Fixes: https://github.com/ceph/ceph-csi-operator/issues/131

**Checklist:**

* [x] **Commit Message Formatting**: Commit titles and messages follow
  guidelines in the [developer
  guide](https://github.com/ceph/ceph-csi-operator/blob/devel/docs/development-guide.md#commit-messages).
* [x] Reviewed the developer guide on [Submitting a Pull
  Request](https://github.com/ceph/ceph-csi-operator/blob/devel/docs/development-guide.md#development-workflow)
* [ ] [Pending release
  notes](https://github.com/ceph/ceph-csi-operator/blob/devel/PendingReleaseNotes.md)
  updated with breaking and/or notable changes for the next major release.
* [x] Documentation has been updated, if necessary.
* [ ] Unit tests have been added, if necessary.
* [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: docs/design/operator.md [hunk 1]
```diff
@@ -98,7 +98,7 @@ spec:
     enableMetadata: true
     grpcTimeout: 100
     snapshotPolicy: auto-detect
-    generateOMapInfo: true
+    generateOMapInfo: false
```
Comment (Madhu-1) [inline docs/design/operator.md:101]: This is a user-facing option to enable the feature, let's revert this change.

Comment (rewantsoni) [reply docs/design/operator.md]: done

**Review by Madhu-1 — APPROVED** [review]
